helper.php

```diff
@@ -0,0 +1,336 @@
+		if (!is_array(@$this->_folder_confs[$folder])){
+			global $conf;
+			$cnf = $conf['plugin']['dw2pdf'];
+			if ($this->get_settingstree()){
+				$cnf = array_merge($cnf,$this->settings_helper->getConf('dw2pdf',$folder));
+			}
+			if (!is_array($cnf)) {$cnf = array();}	// should never happen if the plugin have configs, and this plugin have configs.
+			$this->_folder_confs[$folder] = $cnf;
+		}
+		return $this->_folder_confs[$folder];
+
+	}
+	
+	// CODE MOVED HERE FROM action.php
+	function addExportButton(&$event){
+        if($this->getConf('showexportbutton') && $event->data['view'] == 'main') {
+			// insert button at position before last (up to top)
+			$event->data['items'] = array_slice($event->data['items'], 0, -1, true) +
+				array('export_pdf' => 
+						  '<li>'
+						  .'<a ' . $this->_exportButtonHtml()	// this is the htmlcode for the opening a tag's href and onclick attributes.
+						  . '"  class="action export_pdf" rel="nofollow" title="' . $this->getLang('export_pdf_button') . '">'
+						  . '<span>' . $this->getLang('export_pdf_button') . '</span>'
+						  . '</a>'
+						  . '</li>'
+				) +
+				array_slice($event->data['items'], -1, 1, true);
+        }
+    }
+	private function _getExportUrl($type = 'export_pdf',$params = array(),$_id = null,$_rev = null){
+		global $ID, $REV;
+		if ($_id === null && $_rev === null) $_rev = $REV;	// don't set current page's $REV to an overridden $_id;
+		if ($_id === null) $_id = $ID;
+		$params['do'] = $type;
+		if($_rev) {
+			$params['rev'] = $_rev;
+		}
+		return wl($_id)."?".http_build_query($params);
+	}
+	private function _exportButtonHtml(){
+		$href = $this->_getExportUrl(); $onclick="";
+		if ($this->getConf('enable_export_config_popup') && $this->get_settingstree()){	// fall back to normal export button if settingstree can't be loaded.
+			$opts = array(
+				'path' => $this->_getFolder(null),
+				'pluginname' => 'dw2pdf',
+				'token' => getSecurityToken(),
+				'options' => array(
+					'title' => $this->getLang('export_config_title'),
+				),
+				'on_complete' => 'dw2pdf_export',	// this function is implemented in dw2pdf's script.js 
+			);
+			/* 'settingstree_show_export' is implemented in settingstree's script.js, the plugin may not be installed, so we need to check if it's callable.
+			 * It takes care of all the fuss with creating a popup and displaying the settings inside and returns true if the popup can be displayed, false otherwise.
+			 * The onclick takes care to fall back to normal export button (as returning true) if:
+			 *	 1, there is some kind of script or syntax error
+			 *   2, the settingstree plugin is not installed (or the function is not available for some reason)
+			 *   3, if the popup can not be displayed for some reason. (e.g. the on_complete callback is invalid)
+			 * The json_encode takes care of displaying the array with doublequoted strings, but single quotes still needs to be escaped (e.g. from the title).
+			 */
```

Comment:
Mixing the javascript in the php makes it more difficult to understand what is going on.
Please have a look how you could replace this by e.g. javascript that will replace the default behaviour of the link. 
The link has already an own class. If you needs info from PHP, you could also use https://www.dokuwiki.org/devel:javascript#jsinfo to set info per page that can be accessed in javascript.